js/genkit/src/genkit.ts

```diff
@@ -81,19 +91,65 @@ export interface GenkitOptions {
   plugins?: PluginProvider[];
   /** Directory where dotprompts are stored. */
   promptDir?: string;
-  // FIXME: Default model is not currently supported since the switch to non-global registry.
   /** Default model to use if no model is specified. */
-  defaultModel?: {
-    /** Name of the model to use. */
-    name: string | { name: string };
-    /** Configuration for the model. */
-    config?: Record<string, any>;
-  };
+  model?: ModelArgument<any>;
   // FIXME: This will not actually expose any flows. It needs a new mechanism for exposing endpoints.
   /** Configuration for the flow server. Server will be started if value is true or a configured object. */
   flowServer?: FlowServerOptions | boolean;
 }
 
+export interface ExecutablePrompt<
+  I extends z.ZodTypeAny = z.ZodTypeAny,
+  O extends z.ZodTypeAny = z.ZodTypeAny,
+  CustomOptions extends z.ZodTypeAny = z.ZodTypeAny,
+> {
+  (
+    input?: z.infer<I>,
+    opts?: z.infer<CustomOptions>
+  ): Promise<GenerateResponse>;
+
+  /**
+   * Generates a streaming response by rendering the prompt template with given user input and then calling the model.
+   *
+   * @param opt Options for the prompt template, including user input variables and custom model configuration options.
```

Comment:
Add doc for `@param input` ?

BTW, when would I use `stream` versus `generateStream`, and how would the user know? I think the comments read the same right now for both methods.

Reply:
Updated the docs.

so `(..)`/`.stream` are "convenience" shortcuts for `generate`/`generateStream` where you can just provide the inputs (and optionally model config). If the user needs to provide/override more generate options (additional tools, output settings, etc) then they can use the more verbose `generate`/`generateStream`.